src/utils.ts

```diff
@@ -0,0 +1,26 @@
+import urlTemplate from "url-template";
+
+export function memoize<K, V>(fn: (_key: K) => V): (_key: K) => V {
+	const cache = new Map<K, V>();
+
+	return (key: K) => {
+		if (cache.has(key)) {
+			return cache.get(key) as V;
+		}
+
+		const val = fn(key);
+		cache.set(key, val);
+
+		return val;
+	};
+}
+
+interface Expander {
+	expand(_parameters: unknown): string;
+}
+
+type Parse = (_pathTemplate: string) => Expander;
```

Comment:
The _ in the type param seems unnecessary

```suggestion
interface Expander {
	expand(parameters: unknown): string;
}

type Parse = (pathTemplate: string) => Expander;
```